### PR TITLE
mpvScripts.youtube-chat: init at unstable-2024-06-08

### LINKED ISSUE #264549: Package request: mpv-youtube-chat
**Project description**

From the [readme](https://github.com/BanchouBoo/mpv-youtube-chat):

> Display chat replays overlayed on top of the video for past Youtube livestreams and premiers using [yt-dlp](https://github.com/yt-dlp/yt-dlp)

**Metadata**

* homepage URL: https://github.com/BanchouBoo/mpv-youtube-chat
* source URL: https://github.com/BanchouBoo/mpv-youtube-chat
* license: mit
* platforms: any

### PR DESCRIPTION
**Pull request by FliegendeWurst**
## Description of changes

Package of mpv script: https://github.com/BanchouBoo/mpv-youtube-chat

Fixes #264549
cc @l0b0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/video/mpv/scripts/mpv-youtube-chat.nix
Comment (eclairevoyant) [inline pkgs/applications/video/mpv/scripts/mpv-youtube-chat.nix:3]:
```suggestion
stdenvNoCC.mkDerivation {
```

File: pkgs/applications/video/mpv/scripts/mpv-youtube-chat.nix
Comment (eclairevoyant) [inline pkgs/applications/video/mpv/scripts/mpv-youtube-chat.nix:11]:
```suggestion
    hash = "sha256-22rTiUX6gYR7iknKLoKdBLhY8PMtXIIBLnckt52Sf2E=";
```

File: pkgs/applications/video/mpv/scripts/mpv-youtube-chat.nix
Comment (eclairevoyant) [inline pkgs/applications/video/mpv/scripts/mpv-youtube-chat.nix:17]:
```suggestion
      "opts['yt-dlp-path'] = '${lib.getExe yt-dlp}'"
```

File: pkgs/applications/video/mpv/scripts/mpv-youtube-chat.nix
Comment (eclairevoyant) [inline pkgs/applications/video/mpv/scripts/mpv-youtube-chat.nix:35]: Please set this to yourself

File: pkgs/applications/video/mpv/scripts/mpv-youtube-chat.nix
Comment (eclairevoyant) [inline pkgs/applications/video/mpv/scripts/mpv-youtube-chat.nix:31]: Desc should start with capital letter
https://nixos.org/manual/nixpkgs/unstable/#var-meta-description

**Review by eclairevoyant — DISMISSED** [review]

**Comment by FliegendeWurst** [comment]
Thank you for the review @eclairevoyant. I incorporated your suggestions.

**Review by l0b0 — APPROVED** [review]
Nice, thank you!

**Review by nicoonoclaste — CHANGES_REQUESTED** [review]

**Comment by uninsane** [comment]
@FliegendeWurst i'm happy to test & merge this for you if you get a chance to address the feedback above

File: pkgs/applications/video/mpv/scripts/mpv-youtube-chat.nix
Comment (uninsane) [inline pkgs/applications/video/mpv/scripts/mpv-youtube-chat.nix:20]:
```suggestion
  scriptPath = "youtube-chat";
```

otherwise it errors at runtime `Cannot find main.* for any supported scripting backend in: /nix/store/77kan7dhygd8msbf63kaqfwb7d2axi94-mpv-youtube-chat-unstable-2024-06-08/share/mpv/scripts/.`

**Comment by uninsane** [comment]
tested with:
```sh
$ nix-shell -p 'mpv.override { scripts = [ mpvScripts.youtube-chat ]; }' --command "mpv https://www.youtube.com/watch?v=Ach8lgF6-AY"
```
then hit ``` ` ``` and type `script-binding load-chat`, you should see a diagnostic message and a few seconds later it'll render chat messages in the bottom of the window.

worked for me after applying the feedback above. @nbraud can you unblock?

**Review by uninsane — APPROVED** [review]

**Comment by uninsane** [comment]
@nbraud merging is blocked on you: can you please review the latest changes here and update your PR review?

**Review by nicoonoclaste — APPROVED** [review]

**Pull request merged**
